internal/app/auditd/auditd.go

```diff
@@ -0,0 +1,193 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package auditd registers auditd service and logs audit events.
+package auditd
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"sync"
+	"sync/atomic"
+	"syscall"
+	"time"
+
+	"github.com/elastic/go-libaudit/v2"
+	"github.com/elastic/go-libaudit/v2/aucoalesce"
+	"github.com/elastic/go-libaudit/v2/auparse"
+	"gopkg.in/yaml.v3"
+
+	"github.com/siderolabs/talos/internal/app/machined/pkg/runtime"
+)
+
+// Main is an entrypoint to the auditd service.
+func Main(ctx context.Context, _ runtime.Runtime, logWriter io.Writer) error {
+	return Run(ctx, logWriter)
+}
+
+// [TODO]: decide which format we prefer: short summary or detailed YAML/JSON or raw without coalescing
+//
+// supported values: "summary", "yaml", "json", "raw".
+const format = "json"
+
+type streamHandler struct {
+	writer io.Writer
+}
+
+func (s *streamHandler) ReassemblyComplete(msgs []*auparse.AuditMessage) {
+	event, err := aucoalesce.CoalesceMessages(msgs)
+	if err != nil {
+		fmt.Fprintf(s.writer, "failed to coalesce messages: %v", err)
+
+		return
+	}
+
+	switch format {
+	case "summary":
+		sm := event.Summary
+
+		fmt.Fprintf(
+			s.writer,
+			`%s: sequence=%v category=%v type=%v actor=%v/%v action=%v thing=%v/%v how=%v tags=%v`+"\n",
+			event.Timestamp.Format("2006/01/02 15:04:05.000000"), event.Sequence, event.Category, event.Type, sm.Actor.Primary, sm.Actor.Secondary,
+			sm.Action, sm.Object.Primary, sm.Object.Secondary, sm.How, event.Tags,
+		)
+	case "yaml":
+		s.writer.Write([]byte("---\n"))         //nolint:errcheck
+		yaml.NewEncoder(s.writer).Encode(event) //nolint:errcheck
+	case "json":
+		json.NewEncoder(s.writer).Encode(event) //nolint:errcheck,errchkjson
+	}
```

Comment:
This is not a format SELinux tools, or maybe some generic Linux security tools people might want to pipe audit data into would accept. Honestly it looks way less human-readable than Linux default format :(

Reply:
Or maybe I'm doing something wrong that I cannot get raw logs using talosctl

Reply:
I think raw should be closer, and we can adjust it